### ISSUE TITLE
Bug :  From Binding Falis When Using Complex Object 

### ISSUE THREAD

**Issue by Ayman-Elfaki** [FastEndpoints/FastEndpoints#851]
## The Problem

The model binder does not work with requests that have a colloction.

When running the the sample below the PricingTiers colloction is empty.

Here is the steps/sample to reproduce the issue:

 - Program.cs 

``` csharp

using FastEndpoints;

var builder = WebApplication.CreateBuilder(args);

builder.Services.AddFastEndpoints();

var app = builder.Build();

app.UseFastEndpoints();

app.Run();

```
 - CreateProductEndpoint.cs 

``` csharp
public class CreateProductRequest
{

    [FromForm]
    public required ProductDto Product { get; set; }

    public class ProductDto
    {
        public required IFormFile Image { get; set; }
        public required IEnumerable<PricingTierDto> PricingTiers { get; set; }
    }

    public class PricingTierDto
    {
        public required uint Tier { get; set; }
        public required decimal Price { get; set; }
    }
}

public class CreateProductEndpoint : Endpoint<CreateProductRequest>
{

    public override void Configure()
    {
        Post("/products");
        AllowAnonymous();
        AllowFormData();
    }

    public override async Task HandleAsync(CreateProductRequest req, CancellationToken ct)
    {

        if (!req.Product.PricingTiers.Any())
            ThrowError(p => p.Product.PricingTiers, "Pricing Tiers is empty");

        // Some other logic

        await SendOkAsync(ct);
    }
}
```
Here is the request screenshot:
![image](https://github.com/user-attachments/assets/3655b0ed-8943-49d6-a2e7-761dc2831368)

## The Workaround

Introduce a string property in the collection;

 - CreateProductEndpoint.cs 

``` csharp
public class CreateProductRequest
{

    [FromForm]
    public required ProductDto Product { get; set; }

    public class ProductDto
    {
        public required IFormFile Image { get; set; }
        public required IEnumerable<PricingTierDto> PricingTiers { get; set; }
    }

    public class PricingTierDto
    {
       //  workaround
        public required string Hack { get; set; }

        public required uint Tier { get; set; }
        public required decimal Price { get; set; }
    }
}

public class CreateProductEndpoint : Endpoint<CreateProductRequest>
{

    public override void Configure()
    {
        Post("/products");
        AllowAnonymous();
        AllowFormData();
    }

    public override async Task HandleAsync(CreateProductRequest req, CancellationToken ct)
    {

        if (!req.Product.PricingTiers.Any())
            ThrowError(p => p.Product.PricingTiers, "Pricing Tiers is empty");

        // Some other logic

        await SendOkAsync(ct);
    }
}
```
Here is the request screenshot:
![image](https://github.com/user-attachments/assets/1cb6f802-669f-420f-937c-5fcbc0e8a8b2)

### Enviroment

Version : FastEndpoints - 5.32.0

Framework : .NET9

**Comment by dj-nitehawk** [comment]
i can't seem to reproduce the problem.
copy/pasted your code and i get correct request model.
can you check again and if still doesn't work, attach a repro project i can have a look at?

![screen-record-1734629602](https://github.com/user-attachments/assets/5e8769e6-2c7d-4470-afd6-62cdc416d1b5)

![screen-record-1734629800](https://github.com/user-attachments/assets/839e58cc-9d2c-4655-961f-bef208cd17c3)

**Comment by Ayman-Elfaki** [comment]
Thank you for the quick response,

What you are testing is the workaround not the actual bug.
if remove the string property it won't bind the colloction.

**Comment by dj-nitehawk** [comment]
fixed in `v5.32.0.10-beta`
thanks for reporting 🙏

note there's some minor breaking changes in this version due to upgrading to xunit v3.
have a look at the [changelog](https://github.com/FastEndpoints/FastEndpoints/blob/main/Src/Library/changelog.md). shouldn't take more than a few minutes to migrate.
